File: backend/test_zalo_api.py
#!/usr/bin/env python3
"""
Test script for Zalo phone resolution API
"""

import requests
import json

# Test data
test_data = {
    "token": "test_token_123",
    "access_token": "test_access_token_456"
}

# API endpoint
url = "http://localhost:8000/api/v1/zalo/phone"

def test_zalo_phone_api():
    """Test the Zalo phone resolution API"""
    print("🧪 Testing Zalo Phone Resolution API")
    print(f"📍 Endpoint: {url}")
    print(f"📤 Request data: {json.dumps(test_data, indent=2)}")
    
    try:
        response = requests.post(url, json=test_data, timeout=10)
        
        print(f"📥 Response status: {response.status_code}")
        print(f"📥 Response headers: {dict(response.headers)}")
        
        if response.status_code == 200:
            response_json = response.json()
            print(f"✅ Success response: {json.dumps(response_json, indent=2)}")
        else:
            print(f"❌ Error response: {response.text}")
            
    except requests.exceptions.RequestException as e:
        print(f"❌ Request failed: {str(e)}")

if __name__ == "__main__":
    test_zalo_phone_api()

File: fe/test-zalo-api.js
// Test Zalo API trực tiếp từ browser console
// Paste đoạn code này vào Console của Chrome DevTools

async function testZaloAPI() {
  const apiUrl = 'http://localhost:8000/api/v1/zalo';
  
  // Test endpoint GET /test
  console.log('🧪 Testing GET /api/v1/zalo/test...');
  try {
    const testResponse = await fetch(`${apiUrl}/test`);
    const testData = await testResponse.json();
    console.log('✅ GET /test Success:', testData);
  } catch (error) {
    console.error('❌ GET /test Error:', error);
  }

  // Test endpoint POST /phone
  console.log('📱 Testing POST /api/v1/zalo/phone...');
  try {
    const phoneResponse = await fetch(`${apiUrl}/phone`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        token: 'test_token_from_zalo_miniapp',
        access_token: 'test_access_token_from_zalo_miniapp'
      })
    });

    if (phoneResponse.ok) {
      const phoneData = await phoneResponse.json();
      console.log('✅ POST /phone Success:', phoneData);
    } else {
      const errorData = await phoneResponse.text();
      console.log('⚠️ POST /phone HTTP Error:', phoneResponse.status, errorData);
    }
  } catch (error) {
    console.error('❌ POST /phone Network Error:', error);
  }
}

// Chạy test
testZaloAPI();
